src/string.c

```diff
@@ -1485,7 +1485,7 @@ char *parseemail(char *input,	/* string to parse */
       at="@";
 
     if (strchr(input, '@') == NULL && strstr(input, "&#64;") == NULL)
-    	return input;	// nothing to do here
+    	return strsav(input);	// nothing to do here
```

Comment:
This patch is taken out of context so it's hard to tell to which other part of the code it is related to  Rather than sending the correction, could you resubmit this patch with its other associated code changes?. (only for the issue this one is solving).